libraries/chain/controller.cpp

```diff
@@ -1920,8 +1918,11 @@ struct controller_impl {
          auto branches = fork_db.fetch_branch_from( new_head->id, head->id );
 
          if( branches.second.size() > 0 ) {
-            for( auto itr = branches.second.begin(); itr != branches.second.end(); ++itr ) {
+            // forked branches are in reverse order of how they were applied, signal callback in the order applied
+            for( auto ritr = branches.second.rbegin(), rend = branches.second.rend(); ritr != rend; ++ritr ) {
+               const block_state_ptr& bsptr = *ritr;
                pop_block();
+               forked_trxs_cb( bsptr->trxs );
```

Comment:
Call `forked_trxs_cb` on `branches.second` once, rather than calling the callback for each block.

Reply:
Also check to see if `forked_trxs_cb` converts to true (i.e contains valid target) before calling.